docs/data/base/components/number-field/number-field.md

```diff
@@ -0,0 +1,235 @@
+---
+productId: base-ui
+title: React NumberField component and hook
+githubLabel: 'component: number-field'
+components: NumberField, NumberFieldGroup, NumberFieldInput, NumberFieldIncrement, NumberFieldDecrement, NumberFieldScrubArea, NumberFieldScrubAreaCursor
+hooks: useNumberField
+waiAria: https://www.w3.org/WAI/ARIA/apg/patterns/spinbutton/
+---
+
+# Number Field
+
+<p class="description">The Number Field component provides users with a field for number values, with stepper buttons and a scrub area to increment or decrement the value.</p>
+
+{{"component": "modules/components/ComponentLinkHeader.js", "design": false}}
+
+{{"component": "modules/components/ComponentPageTabs.js"}}
+
+## Introduction
+
+A number field is a UI element that accepts numeric values from the user. `NumberField` is a customizable replacement for the native HTML `<input type="number">` that solves some usability and visual issues while enhancing its functionality.
+
+{{"demo": "UnstyledNumberFieldIntroduction", "defaultCodeOpen": false, "bg": "gradient"}}
+
+## Component
+
+```jsx
+import { NumberField } from '@mui/base/NumberField';
+```
+
+### Anatomy
+
+The `NumberField` component is composed of a root component and a group component which contains an input, and optionally, an increment button, decrement button, and a scrub area with a virtual cursor:
+
+```tsx
+<NumberField>
+  <NumberField.Group>
+    <NumberField.Decrement />
+    <NumberField.Input />
+    <NumberField.Increment />
+    <NumberField.ScrubArea>
+      <NumberField.ScrubAreaCursor />
+    </NumberField.ScrubArea>
+  </NumberField.Group>
+</NumberField>
+```
+
+### Custom structure
+
+Use the `render` prop to override the rendered elements with your own components:
+
+```jsx
+<NumberField render={(props) => <MyNumberField {...props} />}>
+  {/* Subcomponents */}
+</NumberField>
+```
+
+All subcomponents accept the `render` prop.
+
+To ensure behavior works as expected:
+
+- **Forward all props**: Your component should spread all props to the underlying element.
+- **Forward the `ref`**: Your component should use [`forwardRef`](https://react.dev/reference/react/forwardRef) to ensure the Number Field components can access the element via a ref.
+
+A custom component that adheres to these two principles looks like this:
+
+```jsx
+const MyNumberField = React.forwardRef(function MyNumberField(props, ref) {
+  return <div ref={ref} {...props} />;
+});
+```
+
+### Value
+
+The `value` prop holds the number value, and `onChange` is called when it updates:
+
+```jsx
+function App() {
+  const [value, setValue] = useState(0);
+  return (
+    <NumberField value={value} onChange={setValue}>
+      <NumberField.Group>
+        <NumberField.Decrement>&minus;</NumberField.Decrement>
```

Comment:
I'm curious, why not `-`?

Reply:
&minus is wider and looks more in proportion with `+` 😅